presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -447,6 +454,7 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         Optional<Symbol> groupIdSymbol = Optional.empty();
         if (groupingSets.size() > 1) {
             groupIdSymbol = Optional.of(symbolAllocator.newSymbol("groupId", BIGINT));
+            symbolAllocator.addGroupIdSymbolForQuerySpecification(node, groupIdSymbol.get());
```

Comment:
Now when I think about it you could just directly call  `handleGroupings` method from `aggregate(...)` method. This way you could just pass the `groupIdSymbol` symbol directly. Up to you